### PR TITLE
Fixes #25794 - rename unit uuid to pulp_id

### PR DESCRIPTION
**Pull request by jlsherrill**
This will be a more generic name with pulp 3
integration

### CODE REVIEW

File: app/controllers/katello/api/v2/content_view_puppet_modules_controller.rb [hunk 1]
```diff
@@ -9,7 +9,6 @@ class Api::V2::ContentViewPuppetModulesController < Api::V2::ApiController
     param :content_view_id, :number, :desc => N_("content view identifier"), :required => true
     param :name, String, :desc => N_("name of the puppet module")
     param :author, String, :desc => N_("author of the puppet module")
-    param :uuid, String, :desc => N_("the uuid of the puppet module to associate")
```
Comment (jlsherrill) [inline app/controllers/katello/api/v2/content_view_puppet_modules_controller.rb]: this param seems to be incorrect and shouldn't be part of #index

File: app/views/katello/api/v2/repositories/base.json.rabl [hunk 2]
```diff
@@ -2,7 +2,7 @@ object @resource
 
 extends 'katello/api/v2/common/identifier'
 
-attributes :pulp_id => :backend_identifier
+attributes :pulp_id => :pulp_id
```
Comment (beav) [inline app/views/katello/api/v2/repositories/base.json.rabl]: does a change like this require a Katello 4.0 release?

File: test/fixtures/models/katello_errata.yml [hunk 3]
```diff
@@ -3,7 +3,7 @@ security:
   errata_type: "security"
   title: "World is ending"
   severity: "Critical"
-  uuid: "partylikeits1999"
+  pulp_id: "partylikeits1999"
```
Comment (beav) [inline test/fixtures/models/katello_errata.yml]: :tada:

File: app/views/katello/api/v2/repositories/base.json.rabl (hunk 2 above)
Comment (jlsherrill) [reply app/views/katello/api/v2/repositories/base.json.rabl]: whoops that was unintended, i will fix.

Comment (beav) [reply app/views/katello/api/v2/repositories/base.json.rabl]: aw man, i was hoping to get a 4.0 out of this PR :smile_cat:

**Comment by beav** [comment]
@jlsherrill it's a big PR but the changes are all pretty simple. I didn't test it but the code looks good to me aside from the one item you are updating.

IMO after unit tests all pass, I have no issue with merging and letting bats find issues. I don't think there's much to test really since its a giant `s/uuid/pulp_id/g`.

**Comment by jlsherrill** [comment]
@beav i think thats a fine plan.  I've still got some manual testing to do, but plan on relying on unit tests for most of the coverage.

**Comment by beav** [comment]
:green_apple: :greenland: :green_heart: :green_salad:

@jlsherrill do you know if :bat: :bat: :bat:  is currently passing? I don't remember offhand where to look

**Comment by beav** [comment]
@jlsherrill no further comment from me, merge when ready

**Comment by jlsherrill** [comment]
@beav bats runs here: https://ci.theforeman.org/job/katello-nightly-rpm-pipeline/

Will merge today after some more testing.

**Comment by jlsherrill** [comment]
Did some quick testing of syncing, uploading, various cv actions.  Merging!

**Pull request merged**